mne/viz/backends/_pyvista.py

```diff
@@ -629,10 +627,13 @@ def scalarbar(self, source, color="white", title=None, n_labels=4,
             self.plotter.add_scalar_bar(**kwargs)
 
     def show(self):
+        from .renderer import MNE_3D_BACKEND_TESTING
         self.figure.display = self.plotter.show()
         if hasattr(self.plotter, "app_window"):
-            with self.ensure_minimum_sizes():
-                self.plotter.app_window.show()
+            self._pre_ensure_minimum_sizes()
+            self.plotter.app_window.show()
+            if not MNE_3D_BACKEND_TESTING:
```

Comment:
I think it's dangerous to special-case this. It means things like `test_screenshot` might "work" in that they return correct sizes there but will not return correct sizes when people actually use the code in practice. So -1 on this as a solution, hopefully this is just for temporary testing to see if it's causing the problem

Reply:
Honestly I was expecting failures since `post_ensure_minimum_sizes` is disabled for testing.

Reply:
I wouldn't expect failures, the `pre` basically does a `setMinimumSize` which will force Qt to make the widget the correct size (and never be resized smaller than that, only ever bigger). The `post` is the part that's critical, because it restores full resizability to the widget, allowing the *user* to resize it to be smaller if they want while (ideally) keeping it the correct size while disabling the minimum size.

Reply:
This will be tricky to fix then.

Reply:
Basically, I think that the segfault is caused by `process_events` in the post function.

Reply:
I'll push again to verify at least that

Reply:
Because it calls `paintEvent`? We could for example monkey-patch `paintEvent` of `plotter` or whatever via a context manager to be a no-op during those `process_events` calls.

Reply:
> Because it calls `paintEvent`?

I'm not sure yet but I think it's because `paintEvent` is called before `show` is properly done. Considering how slow those calls are on `MacOS`, it could be possible.